sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobChangefeedPagedFlux.java

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.changefeed;
+
+import com.azure.core.util.IterableStream;
+import com.azure.core.util.paging.ContinuablePagedFlux;
+import com.azure.storage.blob.changefeed.implementation.models.BlobChangefeedEventWrapper;
+import com.azure.storage.blob.changefeed.implementation.models.ChangefeedCursor;
+import com.azure.storage.blob.changefeed.models.BlobChangefeedEvent;
+import reactor.core.CoreSubscriber;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+
+/**
+ * Implementation of {@link ContinuablePagedFlux} for Changefeed where the continuation token type is {@link String},
+ * the element type is {@link BlobChangefeedEvent}, and the page type is {@link BlobChangefeedPagedResponse}.
+ */
+public final class BlobChangefeedPagedFlux extends ContinuablePagedFlux<String, BlobChangefeedEvent,
+    BlobChangefeedPagedResponse> {
+
+    private final Changefeed changefeed;
+    private final Integer defaultPageSize = 5000;
+
+    /**
+     * Creates an instance of {@link BlobChangefeedPagedFlux}.
+     *
+     * @param changefeed {@link Changefeed}
+     */
+    BlobChangefeedPagedFlux(Changefeed changefeed) {
+        this.changefeed = changefeed;
+    }
+
+    @Override
+    public Flux<BlobChangefeedPagedResponse> byPage() {
+        return byPage(null, defaultPageSize);
+    }
+
+    @Override
+    public Flux<BlobChangefeedPagedResponse> byPage(String continuationToken) {
+        return byPage(continuationToken, defaultPageSize);
+    }
+
+    @Override
+    public Flux<BlobChangefeedPagedResponse> byPage(int preferredPageSize) {
+        return byPage(null, preferredPageSize);
+    }
+
+    @Override
+    public Flux<BlobChangefeedPagedResponse> byPage(String continuationToken, int preferredPageSize) {
+
+        if (continuationToken != null) {
+            return Flux.error(new UnsupportedOperationException("continuationToken not supported. Use "
+                + "client.getEvents(String) to pass in a cursor."));
+        }
```

Comment:
I think we need to have a broader discussion about this but the reason I don't want to make that an option is because I don't want users to think they have to provide a continuationToken to get the next page in changefeed (the implementation by default has the ability to get every page lazily). In changefeed the cursor is just used to initialize the changefeed object correctly as opposed to continually be fed to get the next page (this is what users are most used to doing). Does that make sense?_

Reply:
which is not far away from one of applications of continuation token.
normally user doesn't touch it. however, if the enumeration is pretty long they might be saving the token when error occur and just use it later to resume - i.e. use saved token to initialize iterable and let it continue without handholding the token.

If we choose to overload the meaning of continuation token then we need to make sure to add javadoc explaining how this work here.

Btw. looks like in .NET PR continuationToken == cursor. See 
https://github.com/Azure/azure-sdk-for-net/pull/11692/files#diff-c69bdde9b907d209a3c6315196fdb137
and
https://github.com/seanmcc-msft/azure-sdk-for-net/blob/feature/storage/changeFeed/sdk/storage/Azure.Storage.Blobs.ChangeFeed/README.md#resume-with-cursor

Reply:
discussed offline, we will throw for now and add in functionality later if required